tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -265,6 +268,70 @@ def test_zipkin_reporter_for_v2_engine(self):
         v2_span_name_part, trace
         ))
 
+  def run_mock_server(self, bin_name, arguments):
+    args = ["../../../build-support/bin/native/cargo.sh", "run", "--release", "--bin", bin_name]
+    if len(arguments) > 0:
+      args = args + ["--"] + arguments
+    process = subprocess.Popen(args, cwd="src/rust/engine", stdin = subprocess.PIPE, stdout = subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines = True)
+    port = None
+    while port is None:
+      match = re.search("localhost:([0-9]+)", process.stdout.readline())
```

Comment:
Could this string be a parameter? I can see someone modifying the servers and changing this output, and thus leading to weird errors.

Reply:
I'm not sure I agree. I agree it's hard-coded and flakey, but because of it, I probably prefer to keep all the hard-coded flakeyness in the same place so that if it changes, whoever is investigating this test failure doesn't have to go and check the caller etc. We also have hard-coded path to `cargo.sh` in here and the fact that the processes can be terminated by sending "\n" to stdin. I'd like to keep all of this nastyness as contained as possible.